editoast/src/views/rolling_stocks/mod.rs

```diff
@@ -74,36 +76,53 @@ async fn get(db_pool: Data<DbPool>, path: Path<i64>) -> Result<Json<RollingStock
 
 #[derive(Debug, Clone, Deserialize, Serialize)]
 pub struct RollingStockForm {
-    #[serde(flatten)]
-    pub common: RollingStockCommon,
+    pub name: String,
+    pub effort_curves: EffortCurves,
+    pub base_power_class: String,
+    pub length: f64,
+    pub max_speed: f64,
+    pub startup_time: f64,
+    pub startup_acceleration: f64,
+    pub comfort_acceleration: f64,
+    pub gamma: Gamma,
+    pub inertia_coefficient: f64,
+    pub features: Vec<String>,
+    pub mass: f64,
+    pub rolling_resistance: RollingResistance,
+    pub loading_gauge: String,
+    pub power_restrictions: Option<JsonValue>,
+    #[serde(default)]
+    pub energy_sources: Vec<EnergySource>,
+    pub electrical_power_startup_time: Option<f64>,
+    pub raise_pantograph_time: Option<f64>,
     pub locked: Option<bool>,
     pub metadata: RollingStockMetadata,
 }
 
 impl From<RollingStockForm> for RollingStockModel {
     fn from(rolling_stock: RollingStockForm) -> Self {
```

Comment:
Since `into_rolling_stock_model` required an `i64` for the id, it will be weird to provide a fake one to replace it by `None` right after calling the function.